n3fit/src/n3fit/vpinterface.py

```diff
@@ -78,6 +78,20 @@ def xfxQ(self, x, Q, n, fl):
             )
         return self.grid_values([fl], [x]).squeeze()[n]
 
+    def _register_photon(self, xgrid):
+        """If the PDF models contain photons, register the xgrid with them"""
+        try:
+            for m in self._lhapdf_set:
+                pl = m.get_layer("add_photon")
+                pl.register_photon(xgrid)
+                # Recompile the model if necessary
+                # (relies on the caching mechanism inside the add_photon layer)
+                if not pl.built:
+                    m.compile()
+        except ValueError:
+            # There's no photon I guess
```

Comment:
I am a bit apprehensive about this. It is catching a too general exception in a complicated piece of code (in that it calls all sorts of other things). Ideally it would be refined to test the exact behavior that may trigger it.

Reply:
I agree.  The check should be exactly at `get_layer`

Also, I think this would fail with more than one replica being fitted (since a `_r` might get added to every extra replica).

A possible solution is to use `m.get_layer_re("add_photon")` which will match `add_photon` but also `add_photon_1`, `add_photon_2`, etc. If there are no matches, `get_layer_re` will return an empty list so it also allows for an explicit check.